### DISCUSSION TITLE
save/show favorited entries

### DISCUSSION THREAD

**Discussion by pathly-dev** [statamic/cms#5289] [Q&A]
Hi there,

I'm not sure, if there is any statamic solution to save the users favorite entries?
I have a blog and the user can save entries as their favorites.
I haven't found anything so far, so I save the favorited id's in an array to the localstorage.
But now I'm not sure how to show them respectively filter the collection. I have some other filters that can be set through the frontend and there I filter that like 

```
{{ collection:blog :filter1:is="get:filter1" }}

{{ /collection:blog }}
```

but for the localstorage I'm not sure how to put these in the brackets since it seems like I cannot use javascript in there

```
{{ collection:blog :filter1:is="get:filter1"  }}
  {{ if JSON.parse(localStorage.getItem("favorites")).contains(id) }}

  {{ /if }}
{{ /collection:blog }}
```

**Comment by ryanmitchell** [comment]
If you are using local storage then your if JSON statement should be in script...

```
{{ collection:blog :filter1:is="get:filter1"  }}
  <script>
    if (JSON.parse(localStorage.getItem("favorites")).contains({{ id }})) {
       // do something
    }
  </script>
{{ /collection:blog }}
```
